packages/office-ui-fabric-react/src/utilities/positioning/positioning.ts

```diff
@@ -440,7 +442,15 @@ function _finalizeElementPosition(
   const hostRect: Rectangle = _getRectangleFromElement(hostElement);
   const elementEdge = coverTarget ? targetEdge : targetEdge * -1;
   const elementEdgeString = RectangleEdge[elementEdge];
-  const returnEdge = alignmentEdge ? alignmentEdge : _getFlankingEdges(targetEdge).positiveEdge;
+  let returnEdge = alignmentEdge ? alignmentEdge : _getFlankingEdges(targetEdge).positiveEdge;
+
+  // if the element is closer to one side of the bounds than the other, flip the return edge to ensure it grows inwards
+  if (bounds) {
+    const flankingEdges = _getFlankingEdges(targetEdge);
+    const posDiff = _getRelativeEdgeDifference(elementRectangle, bounds, flankingEdges.positiveEdge);
+    const negDiff = _getRelativeEdgeDifference(elementRectangle, bounds, flankingEdges.negativeEdge);
+    returnEdge = Math.abs(posDiff) >= Math.abs(negDiff) ? flankingEdges.negativeEdge : flankingEdges.positiveEdge;
```

Comment:
These aren't really posDiff and negDiff, since you really want to test, is the current alignment edge closer to it's side, than the other possible edge, and if so, return that other value.